### ISSUE TITLE
install tooling  tries to install Titanium CLI and SDK then fails

### ISSUE THREAD

**Issue by jdfabs** [tidev/vscode-titanium#1246]
Type: <b>Bug</b>

Can't find any way to make the tooling work.

Extension version: 1.3.0
VS Code version: Code 1.95.3 (f1a4fb101478ce6ec82fe9627c43efbf9e98c813, 2024-11-13T14:50:04.152Z)
OS version: Linux x64 6.9.3-76060903-generic
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|13th Gen Intel(R) Core(TM) i5-13600KF (20 x 5099)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off<br>webnn: disabled_off|
|Load (avg)|0, 1, 1|
|Memory (System)|31.18GB (19.57GB free)|
|Process Argv|--crash-reporter-id 301ce5f7-725d-46c9-bf0b-902cf21c21a3|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|pop|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|pop|
|XDG_SESSION_TYPE|x11|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyonecf:30548226
a9j8j154:30646983
962ge761:30959799
pythonnoceb:30805159
asynctok:30898717
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
01bff139:31013167
dvdeprecation:31068756
dwnewjupyter:31046869
2f103344:31071589
nativerepl2:31139839
pythonrstrctxt:31112756
nativeloc1:31185841
cf971741:31144450
iacca1:31171482
notype1cf:31157160
5fd0e150:31155592
dwcopilot:31170013
stablechunks:31184530

```

</details>

**Comment by ewanharris** [comment]
Apologies for that @jdfabs, if you try and install the tooling manually does it work?

1. Install [Node.js](https://nodejs.org/en/download/package-manager)
2. Install Titanium and Alloy CLI `npm install titanium alloy -g`
3. Install the latest Titanium SDK `ti sdk install`

**Comment by jdfabs** [comment]
Hello,

I went on a journey for the last hour or so! And I literally just solved my
issue.
the solution ended up being:
in a titanium project settings.json adding:

"terminal.integrated.env.linux": {
  "PATH": "/home/skipper/.nvm/versions/node/v18.x.x/bin:${env:PATH}"}
that made the tidev addon recognise the CLI and SDK properly.

any questions about this issue/solutions please let me know

thanks for your attention in general,
João Silva

**Comment by jdfabs** [comment]
Yes, both ti and alloy work from linux terminal, the project files were
created with ti create. And yes, nvm is on the PATH, with the correct
version being referenced.

Another thing that might be of note is that after I added the path to the
settings.json on the first project (created with ti create)
I deleted the project and rebooted, and when loaded vscode tidev was still
working as expected.

One of the things that I tried doing was to go into vscode the
extension settings and manually set the path, which I found that was not
actually a setting, maybe that's an useful thing to add to the extension.

Once more, thanks for the attention,
João Silva
